src/main/scala-2.13.0-M5/scala/swing/MutableShim.scala

```diff
@@ -0,0 +1,41 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2007-2018, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package scala.swing
+
+import scala.collection.mutable
+
+/**
+ * Shims for changing collections API.
+ */
+protected[swing] trait MutableBufferShim[A] extends mutable.Buffer[A] {
+  override def insert(i: Int, a: A) = insertAt(i, a)
+  protected def insertAt(i: Int, a: A): Unit
+
+  override def insertAll(n: Int, elems: IterableOnce[A]): Unit = insertAllImpl(n, elems.iterator)
```

Comment:
Why do we need `insertAllImpl`? It is only ever used in this call. Could it just be inlined? If not, it should probably be `protected`?